### ISSUE TITLE
Add branching model support to Gitea

### ISSUE THREAD

**Issue by sgabenov** [go-gitea/gitea#26164]
### Feature Description

Like "Bitbucket" server would be great to support brunching models. 
I think there could be a config for repo, that can specify what branch models can be used in the repo and give UI interface to select prefixes, when creating branches from UI.
Also, there could be some "POLICY CHECKS", so users could name/create/push branches only from specified branch model that is set to repo.
 

### Screenshots

![image2015-7-28+10_40_30](https://github.com/go-gitea/gitea/assets/15741789/26903e42-2c9f-46d8-bc70-bd8cbd377cb3)

**Comment by JakobDev** [comment]
What exactly is the difference between this and just creating a branch called `feature/foo`?

**Comment by sgabenov** [comment]
> What exactly is the difference between this and just creating a branch called `feature/foo`?

The difference is the ability to set automation on this. If we enable Policy to allow only to create "feature/ bugfix/ hotfix" in repo config, the dropdown meny will allow to use only predefined prefixes? that will be listed. In other cases user will be guessing what is allowed to create\what not. 
But this is only actual if additional functionality for Gitea will be added in future releases, where we can specify what brunch model (prefixes) is allowed in the current repo.

**Comment by delvh** [comment]
So, your problem sounds like some sort of reverse branch-protection to me:
Instead of protecting the branch you're pushing to, you disallow branches that may not be created?

**Comment by Stuw** [comment]
This feature request looks like a user friendly way to force a naming convention in a repo.

**Comment by DaKaLKa** [comment]
Maybe Take a Look at Sourcetree smart branching - https://blog.sourcetreeapp.com/2012/08/01/smart-branching-with-sourcetree-and-git-flow/
Sourcetree optionally use a gitflow based branching strategy which sounds similar and helps unexperienced users using git in a common way. (E.g. it automatically merges hotfixes to dev , release and prod branches)

The mentioned Feature in bitbucket can also bei triggered e.g. from Jira - so you can simply create a branch from a Ticket - the Name of the brabch is automatically generated from the issue Ticket ID + Name.

**Comment by stuzer05** [comment]
Yes, this is a very needed feature. Also, a "create branch" button on issue's page would be super awesome

**Comment by MuhmdHsn313** [comment]
This feature is highly valuable, especially when it has support for git-flow, making repository management more straightforward for teams. Are there any updates regarding its status?
